subsys/net/lib/tftp/tftp_client.c

```diff
@@ -53,6 +53,61 @@ static size_t make_request(uint8_t *buf, int request,
 	return ptr - (char *)buf;
 }
 
+/*
+ * Send an Error Message to the TFTP Server.
+ */
+static int send_data(int sock, uint32_t block_no, size_t data_size, uint8_t *data_buffer)
+{
+	int ret;
+	int tx_count = 0;
+	struct pollfd fds = {
+		.fd     = sock,
+		.events = ZSOCK_POLLIN,
+	};
+
+	LOG_DBG("Client send data: block no %u, size %u", block_no, data_size);
+
+	/* Send data and poll for ACK response */
+	sys_put_be16(DATA_OPCODE, data_buffer);
+	sys_put_be16(block_no, data_buffer + 2);
+	do {
+		if (tx_count > TFTP_REQ_RETX) {
+			LOG_ERR("No more retransmits. Exiting");
+			return TFTPC_RETRIES_EXHAUSTED;
+		}
+		ret = send(sock, data_buffer, data_size + TFTP_HEADER_SIZE, 0);
```

Comment:
`sendto()` should be used, there was no `connect()` on the socket.

Reply:
connect() has been done in send_request(), becoming connection-oriented UDP

Reply:
Ah, ok indeed, quite hidden though.